janus_server/src/datastore.rs

```diff
@@ -78,7 +81,8 @@ impl Transaction<'_> {
 
     // This is pub to be used in integration tests
     #[doc(hidden)]
-    pub async fn put_task(&self, task_id: TaskId) -> Result<(), Error> {
+    pub async fn put_task(&self, task: &TaskParameters) -> Result<(), Error> {
+        // TODO: interpolate values from `task` into prepared statement
```

Comment:
I had trouble doing this because I couldn't figure out how to render a `task::Vdaf` into the `VDAF_IDENTIFIER` SQL enum, as well as creating a Postgres `INTERVAL` from `Duration`.

Reply:
For the enum, we likely want to use [ToSql/FromSql derive macros](https://docs.rs/postgres-types/latest/postgres_types/#enums) along with [some renaming attribute macros](https://docs.rs/postgres-types/latest/postgres_types/#naming) to handle different case conventions. I don't see any conversions for `INTERVAL`, so we may have to implement `ToSql`/`FromSql` on a newtype of Duration ourselves, for `Type::INTERVAL`.

Reply:
Ah, looking at the rust-postgres issue tracker, we actually may want to change what we're using in the database. `INTERVAL` has multiple separate fields (it takes up 16 bytes total) and it can represent ambiguous things like "one year plus one month". Maybe we write `ToSql`/`FromSql` for our own purposes to be strict enough that every field for months and upwards must always be zero, or maybe we make a DOMAIN  of BIGINT to represent a duration in seconds, and use that.

Reply:
Yes, and at that point we have to pull in crate `postgres_types`. I decided to "cut my losses" and stop this PR from getting too big.

Reply:
https://www.postgresql.org/docs/current/datatype-datetime.html:
> interval internally stores only three integer units (months, days, microseconds), fractional units must be spilled to smaller units.

!! I support changing `min_batch_duration` to e.g. a BIGINT counting a number of seconds.

Reply:
Approving this PR, I suppose the datastore changes don't need to go here.

Reply:
I made a note about `min_batch_duration` in #22.